src/tests/compiler/class/classDeclarationTests.ts

```diff
@@ -1347,4 +1348,64 @@ class Child extends Mixin(Base) {}
             doTest("class I {}\n\nclass J {}\n\nclass K {}", 1, "class I {}\n\nclass K {}");
         });
     });
+
+    describe(nameof<ClassDeclaration>(d => d.getStructure), () => {
+        function doTest(code: string, expectedStructure?: ClassDeclarationStructure) {
+            const {descendant, project} = getInfoFromTextWithDescendant<ClassDeclaration>(code, SyntaxKind.ClassDeclaration);
+            const structure = descendant.getStructure();
+            if (expectedStructure) {
+                expect(Object.assign({}, structure, {methods: undefined, properties: undefined, ctors: undefined})).to.contain(
+                    Object.assign({}, expectedStructure, {methods: undefined, properties: undefined, ctors: undefined}));
+            }
+            const sourceFile2 = project.createSourceFile("__tmp_sourceFile1.ts", "", {overwrite: true});
+            const generatedClassDecl = sourceFile2.addClass(structure);
+            expect(generatedClassDecl.getText().replace(/\s+/gm, "")).equals(descendant.getText().replace(/\s+/gm, ""));
```

Comment:
Good idea to do this text comparison. Do we need to remove the spaces though?

Reply:
I was just thinking about this more. Could we remove the add with text comparisons? That way we're testing the structures directly which would catch any issues with it.